sycl/doc/extensions/proposed/sycl_ext_intel_fpga_kernel_arg_properties.asciidoc

```diff
@@ -0,0 +1,508 @@
+template <typename T, typename PropertyListT>
+struct is_property_key_of<latency_key,
+  annotated_ptr<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<maxburst_key,
+  annotated_ptr<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<wait_request_key,
+  annotated_ptr<T, PropertyListT>> : std::true_type {};
+
+template <typename T, typename PropertyListT>
+struct is_property_key_of<conduit_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<register_map_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<stable_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<buffer_location_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<awidth_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<dwidth_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<read_write_mode_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<latency_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<maxburst_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<wait_request_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+
+} // namespace sycl::ext::oneapi::experimental
+```
+--
+
+[frame="topbot",options="header"]
+|===
+|Property |Description
+
+a|
+[source,c++]
+----
+conduit
+----
+a|
+Directs the compiler to create a dedicated input port on the kernel for the
+input.
+
+a|
+[source,c++]
+----
+register_map
+----
+a|
+Directs the compiler to create a register to store the input as opposed to
+creating a dedicated input port on the kernel.
+
+a|
+[source,c++]
+----
+stable
+----
+a|
+Specifies that the input to the kernel will not change during the execution of
+the kernel. The input can still change after all active kernel invocations have
+finished.
+
+If the input is changed while the kernel is executing, the behavior is
+undefined.
+
+a|
+[source,c++]
+----
+buffer_location<id>
+----
+a|
+Specifies a global memory identifier for the pointer interface.
+
+This property is only meaningful on pointer kernel arguments.
+
+a|
+[source,c++]
+----
+awidth<width>
+----
+a|
+Specifies the width of the memory-mapped address bus in bits. The default is
+set to 64.
```

Comment:
I thought we were considering a narrower width.  If we're unsure, perhaps we should omit mention of a default or say that the default is implementation-defined.